multi_partner_learning.py

```diff
@@ -435,42 +480,57 @@ def aggregate_model_weights(self):
             new_weights.append(list(avg_weights_for_layer))
 
         return new_weights
+    
 
     def build_model_from_weights(self, new_weights):
         """Generate a new model initialized with weights passed as arguments"""
 
         new_model = self.generate_new_model()
         new_model.set_weights(new_weights)
-        new_model.compile(
-            loss=keras.losses.categorical_crossentropy,
-            optimizer="adam",
-            metrics=["accuracy"],
-        )
```

Comment:
Are you sure after a `set_weights()` it is ok to not compile? I thought @RomainGoussault had told me you had to, but I might be wrong.

Reply:
Compile does not impact the weights. You have to compile again if you change the loss or the optimizer for example. We already compiled the model in `generate_new_model()` so we do not have to do it again. 
https://stackoverflow.com/questions/47995324/does-model-compile-initialize-all-the-weights-and-biases-in-keras-tensorflow/47996024

If we want to reinitialize optimizer or loss, we should compile again.